### ISSUE TITLE
Multiple rendering issues for aio in Windows high contrast mode

### ISSUE THREAD

**Issue by jelbourn** [angular/angular#44336]
I've observed the following issues for aio in Windows High Contrast Mode

- [ ] The main header navigation links (Docs, Resources, etc) do not have a focus treatment
![image](https://user-images.githubusercontent.com/838736/144323055-2870c4bc-f820-4028-ba24-bb8641b57baa.png)

- [ ] The search text input does not have a border
![image](https://user-images.githubusercontent.com/838736/144323245-e5011c67-f9a6-4ec1-87a5-877d1df600b2.png)

- [ ] The copy button for code snippets/examples does not have a border (this should probably be a `mat-icon-button`)
![image](https://user-images.githubusercontent.com/838736/144323563-4aac4d6d-48af-4e7a-8f75-a45a18dae08c.png)

- [ ] The position indicator in the table of contents is almost impossible to see
![image](https://user-images.githubusercontent.com/838736/144324176-e1b95775-f888-4c42-ba52-f30262521601.png)

- [ ] The "Suggest edits" icon doesn't have a border (this should also probably be a `mat-icon-button`)
![image](https://user-images.githubusercontent.com/838736/144324426-ab04ce8a-54d4-4e8a-87bd-e61212ac7197.png)

- [ ] There's no separator between an example file header and its content
![image](https://user-images.githubusercontent.com/838736/144324551-3beface0-b7f1-4b17-acb0-5b299b16524f.png)
